ethcore/src/client/client.rs

```diff
@@ -2217,22 +2217,22 @@ impl BlockChainClient for Client {
 }
 
 impl IoClient for Client {
-	fn queue_transactions(&self, transactions: Vec<Bytes>, peer_id: usize) {
+	fn queue_transactions(&self, transactions: Vec<UnverifiedTransaction>, peer_id: usize) {
 		trace_time!("queue_transactions");
 		let len = transactions.len();
 		self.queue_transactions.queue(&self.io_channel.read(), len, move |client| {
 			trace_time!("import_queued_transactions");
-
-			let txs: Vec<UnverifiedTransaction> = transactions
-				.iter()
-				.filter_map(|bytes| client.engine.decode_transaction(bytes).ok())
```

Comment:
this verification is performed by a miner in `import_external_transactions`

Reply:
So we previous ignored invalid rlps and now we return an err in `on_peer_transactions`? This seems like an improvement to me.

Reply:
> So we previous ignored invalid rlps and now we return an err in on_peer_transactions? 

Yes

> This seems like an improvement to me. But maybe it was done on purpose?

No, it was a regression introduced very recently

Reply:
This closure is invoked in a `IoWorker` thread pool and the whole thing was done to prevent blocking networking task for  useless stuff like decoding transactions.